contracts/jobs/JobsManager.sol

```diff
@@ -281,19 +287,51 @@ contract JobsManager is ManagerProxyTarget, IVerifiable, IJobsManager {
 
         // Segment must be eligible for verification
         require(JobLib.shouldVerifySegment(_segmentNumber, claim.segmentRange, claim.claimBlock, verificationRate));
+        // Segment must be signed by broadcaster
+        require(JobLib.validateBroadcasterSig(job.streamId, _segmentNumber, _dataHash, _broadcasterSig, job.broadcasterAddress));
         // Receipt must be valid
-        require(validateReceipt(_jobId, _claimId, _segmentNumber, _dataHash, _transcodedDataHash, _broadcasterSig, _proof));
+        require(JobLib.validateReceipt(job.streamId, _segmentNumber, _dataHash, _transcodedDataHash, _broadcasterSig, _proof, claim.claimRoot));
 
         // Mark segment as submitted for verification
         claim.segmentVerifications[_segmentNumber] = true;
 
         // Invoke transcoding verification. This is async and will result in a callback to receiveVerification() which is implemented by this contract
-        verifier().verify(_jobId, _claimId, _segmentNumber, _dataHash, _transcodedDataHash, this);
+        invokeVerification(_jobId, _claimId, _segmentNumber, _dataStorageHash, _transcodedDataHash);
 
         return true;
     }
 
     /*
+     * @dev Invoke transcoding verification by calling the Verifier contract
+     * @param _jobId Job identifier
+     * @param _claimId Claim identifier
+     * @param _segmentNumber Segment sequence number in stream
+     * @param _dataStorageHash Content addressable storage hash of segment data
+     * @param _transcodedDataHash Keccak256 hash of transcoded segment data
+     */
+    function invokeVerification(
+        uint256 _jobId,
+        uint256 _claimId,
+        uint256 _segmentNumber,
+        string _dataStorageHash,
+        bytes32 _transcodedDataHash
+    )
+        internal
+        returns (bool)
+    {
+        IVerifier verifierContract = verifier();
+
+        uint256 price = verifierContract.getPrice();
+
+        // Send payment to verifier if price is greater than zero
+        if (price > 0) {
+            return verifierContract.verify(_jobId, _claimId, _segmentNumber, jobs[_jobId].transcodingOptions, _dataStorageHash, _transcodedDataHash);
+        } else {
+            return verifierContract.verify.value(price)(_jobId, _claimId, _segmentNumber, jobs[_jobId].transcodingOptions, _dataStorageHash, _transcodedDataHash);
```

Comment:
Since this function is only called by `verify`, which will ensure that a transcoder provided enough payment to cover the price of verification, we can forward that payment to the verifier